scss/_custom-forms.scss

```diff
@@ -419,3 +419,126 @@
     @include border-radius($custom-range-track-border-radius);
   }
 }
+
+// Switch
+//
+// Android style $custom-switch-track-height < $custom-switch-thumb-height
+// iOS style     $custom-switch-track-height >= $custom-switch-thumb-height
+@function custom-switch-thumb-left($checked) {
+  @if ($checked == 1) {
+    @if ($custom-switch-padding-inner > 0) {
+      @return $custom-switch-track-width - $custom-switch-thumb-width - $custom-switch-padding-inner;
+    } @else {
+      @return $custom-switch-track-width - $custom-switch-thumb-width - $custom-switch-padding-inner * 2;
+    }
+  } @else {
+    @if ($custom-switch-padding-inner > 0) {
+      @return $custom-switch-padding-inner;
+    } @else {
+      @return 0;
+    }
+  }
+}
+
+.custom-switch {
+  padding-left: 0;
+  .custom-control-input {
+    ~ .custom-control-label {
+      @if ($custom-switch-padding-inner > 0) {
+        padding-left: $custom-switch-track-width + $custom-switch-padding-outer;
+      } @else {
+        padding-left: $custom-switch-track-width + $custom-switch-padding-outer - $custom-switch-padding-inner * 2;
+      }
+      &::before,
+      &::after {
```

Comment:
No need for this, just add the properties in the `::after` or `::before` selectors. You're increasing the number of selectors with the technique you're using here.